### PR TITLE
feat(crosswalk): resolve ESPN conference labels from the Core v2 group tree

### PR DESCRIPTION
**Pull request by saiemgilani**
## The blocker

`espn_team_directory` emitted **no conference column for any basketball league**, so `wbb_team_crosswalk` shipped `espn_conference` at **0/362** where the R producer ships 361/361. That was the blocker on the WBB crosswalk.

## Where the R producer actually gets it

ESPN's Site v2 `teams` directory genuinely carries no conference — that part of the diagnosis was right. `wehoop::espn_wbb_teams()` reconstructs it by walking the **Core v2 season group tree**:

1. `.../seasons/{year}/types/2/groups/50/children` — the children of group 50 ("NCAA Division I") are the 31 conferences
2. `.../groups/{gid}` — the conference's `name`
3. `.../groups/{gid}/teams` — its member teams

`espn_conference_map()` does the same walk through sdv-py's **already-existing** `espn_{lg}_season_group` / `_children` / `_teams` wrappers. No new endpoint, no codegen change, no new dependency.

## Live 2026 vs the committed R golden

| | live | golden |
|---|---|---|
| `espn_conference` populated | **361/362** | 361/361 |
| distinct conferences | 31 | 31 |
| teams in >1 conference | 0 | — |

`match_method` is **byte-identical to the pre-change baseline** — `fox+bart` 357 / `bart_only` 2 / `fox_only` 2 / `espn_only` 1 — so this change is strictly additive and the `match_method` spread vs the golden is pre-existing live-vs-frozen drift, not introduced here. (Diffed with `eq_missing`, not `filter(~(a == b))`.)

The 20 name differences vs the golden are **ESPN renames since the golden was frozen**: MAAC → "Metro Conference" (13 teams), WAC → "United Athletic Conference" (7). Core v2 and Site v2 `scoreboard/conferences` agree on the new names with **zero disagreement across all 31 groups**, so a fresh R run today produces them too. The single null is West Florida (2697), which live ESPN places in no D-I conference group.

MBB verified live as well (365 teams, 31 conferences) — though the MBB crosswalk stays blocked on KenPom regardless.

## Contract

- A walk that resolves **no teams at all** raises `CrosswalkSourceError` rather than shipping an all-null column.
- A **single** conference that fails to fetch is skipped, matching the R producers' per-item tolerance.
- The join key goes through `str_id`, so an `Int64` team id stringifies as `"52"` and never the float-origin `"52.0"`.
- NBA/WNBA have no group-50 tree and are unchanged (their crosswalks carry no conference column).

## Gates

`ruff check` + `ruff format` clean · `mypy` (both files already on the ratchet) clean · `pytest tests/test_crosswalk_basketball_sources.py tests/test_basketball_crosswalk_parity.py` **68 passed** (6 new) · `generate.py --check` all generated files current.

## Summary by Sourcery

Populate ESPN conference labels for NCAA basketball crosswalks by reconstructing conferences from the Core v2 season group tree and wiring them into the team directory and WBB team crosswalk.

New Features:
- Introduce espn_conference_map to derive ESPN team-to-conference mappings for NCAA seasons via Core v2 season group hierarchies.

Bug Fixes:
- Ensure espn_conference is populated for NCAA WBB teams instead of shipping an all-null conference column.

Enhancements:
- Augment espn_team_directory to resolve conference_name for NCAA leagues when not provided upstream, joining on a canonical string team_id.
- Update WBB team crosswalk documentation to describe conference reconstruction from the Core v2 group tree.

Tests:
- Add unit tests covering the conference group-tree walk, error handling for empty results, tolerance of failing conference groups, non-NCAA league behavior, and correct joining of conference labels into the ESPN team directory.

## Summary by CodeRabbit

- **New Features**
  - Added improved NCAA women’s basketball conference mapping using ESPN season data.
  - Team directories now include conference names when available, with automatic reconstruction when upstream labels are missing.
  - Supports more accurate conference assignments across supported seasons and team identifiers.

- **Bug Fixes**
  - Improved handling of incomplete or unavailable conference data without preventing other teams from loading.
  - Preserved existing conference information when it is already provided by ESPN.

- **Documentation**
  - Updated crosswalk documentation to explain conference-name resolution behavior.

### CODE REVIEW

File: sportsdataverse/_crosswalk_basketball_sources.py [hunk 1]
```diff
@@ -226,9 +370,16 @@ def espn_team_directory(league: str, season: Optional[int] = None, **kwargs: Any
...
+            return out.with_columns(raw[candidate].cast(pl.Utf8).alias("conference_name"))
+
+    if season is None or _ncaa_group_accessors(league) is None:
+        return out
+    # Not **kwargs: those are the *teams* accessor's, and the group endpoints
+    # take a different parameter set.
+    conferences = espn_conference_map(league, int(season))
+    # Join on Utf8 both sides -- as_str_id keeps a numeric id off the float
+    # path, so an Int64 team_id stringifies as "123" and never "123.0".
+    return out.with_columns(str_id(out, "team_id")).join(conferences, on="team_id", how="left")
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/_crosswalk_basketball_sources.py]: Valid, fixed in d90a2594.

Two halves, because "infers Utf8" was the real weakness:

1. `espn_conference_map` now **pins** the dtype on the populated return path
   (`schema={"team_id": pl.Utf8, "conference_name": pl.Utf8}` on the
   constructor) instead of relying on inference happening to agree with the
   typed-empty path's declared schema. Both return paths now hand back the
   same declared dtype.
2. `espn_team_directory` materializes the normalized frame and asserts before
   the join, with a message naming both sides:

```python
out = out.with_columns(str_id(out, "team_id"))
assert out.schema["team_id"] == conferences.schema["team_id"] == pl.Utf8, (
    f"team_id join-key dtype mismatch: directory={out.schema['team_id']}, "
    f"conferences={conferences.schema['team_id']} (both must be {pl.Utf8})"
)
return out.join(conferences, on="team_id", how="left")
```

No cast was added on either side -- `str_id`/`as_str_id` already routes a
float-origin id through `Int64` first, so the "123.0" paper-over stays off the
table.

Re-verified live afterwards (`wbb_team_crosswalk(season=2026)`): 362 rows,
`espn_conference` 361/362 across 31 conferences, and an `eq_missing` cell diff
against the pre-change run reports **0 differing cells across all 18 columns**.

File: sportsdataverse/wbb/wbb_crosswalk.py [hunk 2]
```diff
@@ -369,9 +369,12 @@ def wbb_team_crosswalk(
...
     Note:
-        ``espn_conference`` is null unless the ESPN frame carries a
-        ``conference_name`` column -- sdv-py's ``espn_wbb_teams()`` does not
-        ship conference labels.
+        sdv-py's ``espn_wbb_teams()`` ships no conference labels, so
+        ``espn_conference`` is reconstructed from ESPN's Core v2 season group
+        tree (see
+        :func:`~sportsdataverse._crosswalk_basketball_sources.espn_conference_map`),
+        the same walk the R producer does. A pre-fetched ``espn`` frame that
+        already carries ``conference_name`` is used as-is.
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/wbb/wbb_crosswalk.py]: Valid, fixed in d90a2594 -- and the same wrong claim was live in `mbb`, so it
was fixed there too.

`wbb_team_crosswalk` has no `espn` parameter; the directory is always fetched
in-function. The Note now says that explicitly and describes the actual
preservation path (`espn_team_directory` keeps `conference_name` when the
upstream ESPN response carries one, and skips the group walk in that case).
`CrosswalkSourceError` propagated out of `espn_conference_map` is now in
`Raises:`, with the note that the ESPN directory -- unlike Fox/Torvik -- is not
opt-out-able.

`mbb_team_crosswalk` calls `espn_team_directory("mbb", season=season)` on the
same code path (`mbb_crosswalk.py:488`), so its Note ("`espn_conference` is
null unless the ESPN frame carries a `conference_name` column") had gone stale
in exactly the same way. Both leagues now carry identical, correct text.

`nba`/`wnba` were checked and need no change: `_ncaa_group_accessors` returns
`None` for them, so `espn_team_directory` returns before the walk, and neither
docstring makes a conference claim.

File: tests/test_crosswalk_basketball_sources.py [hunk 3]
```diff
@@ -412,3 +414,134 @@ def test_team_crosswalk_supplied_ratings_frame_needs_no_provider_module(
...
+    assert out["team_id"].to_list() == ["52"]
+
+
+def test_espn_conference_map_is_empty_for_non_ncaa_leagues(monkeypatch: pytest.MonkeyPatch) -> None:
+    """NBA/WNBA have no group-50 tree; their crosswalks carry no conference."""
+    for league in ("nba", "wnba"):
+        out = espn_conference_map(league, 2026)
+        assert out.height == 0
+        assert out.columns == ["team_id", "conference_name"]
+
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/test_crosswalk_basketball_sources.py:502]: Valid, fixed in d90a2594 -- same class as the `_TEAMS_SCHEMA` finding on #349.

```python
out = espn_conference_map(league, 2026)
assert out.height == 0
# Full schema, not just names: a Null-dtyped ``team_id`` is an unusable
# join key that a columns-only assertion would wave through.
assert dict(out.schema) == {"team_id": pl.Utf8, "conference_name": pl.Utf8}
```

Paired with the producer-side fix in the sibling thread: the populated return
path now pins the same dtypes rather than inferring them, so this assertion
and `test_espn_conference_map_walks_the_group_tree`'s existing schema check
together pin both return paths of the function.

53 tests green in `tests/test_crosswalk_basketball_sources.py` +
`tests/test_id_conventions.py`.
